File: src/main/java/com/cfgglobal/test/service/RoleService.java
package com.cfgglobal.test.service;

import com.cfgglobal.test.domain.Role;
import com.cfgglobal.test.domain.RolePermission;
import com.cfgglobal.test.service.base.BaseService;
import io.vavr.collection.List;
import org.springframework.stereotype.Service;


@Service
public class RoleService extends BaseService<Role, Long> {

    public Role removeEmptyRules(Role role) {
        java.util.List<RolePermission> rolePermissions = role.getRolePermissions();
        return role.setRolePermissions(List.ofAll(rolePermissions)
                .filter(e -> e.getRules() != null && !e.getRules().isEmpty())
                .asJava());
    }
}

File: src/main/kotlin/com/cfgglobal/test/service/RoleService.kt
package com.cfgglobal.test.service

import com.cfgglobal.test.domain.Role
import com.cfgglobal.test.service.base.BaseService
import org.springframework.stereotype.Service


@Service
class RoleService : BaseService<Role, Long>() {

    fun removeEmptyRules(role: Role): Role {
        val rolePermissions = role.rolePermissions
        role.rolePermissions.addAll(rolePermissions.filter { (_, rules) -> !rules.isEmpty() })
        return role

    }
}
